### PR TITLE
Add the latest instructions of seting up dev env using Docker.

### PR DESCRIPTION
**Pull request by siqidong**
Update the developer guide.

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -35,10 +35,60 @@ These jobs use GCP project `cloudesf-testing`.
...
-We recommend starting this setup using only the command line at first.
+There are two ways to setup the development environment for this project: the Docker mode and
+the native mode. Due to the frequent version changes of software dependecies, it is recommended to
+setup your dev env in Docker mode.
 Later, we will walk through how to use Intellij and CLion IDEs for the project.
 
-### Install Golang
+### Docker Mode Setup
+
+#### Run Docker Instance
```
Comment (shuoyang2016) [inline DEVELOPER.md:45]: Great. I like this mode:)

**Review by shuoyang2016 — APPROVED** [review]
Thanks!

File: DEVELOPER.md (hunk 1 above)
Comment (siqidong) [reply DEVELOPER.md:45]: Thank you Shuo for your quick review!

**Review by TAOXUY — APPROVED** [review]

File: DEVELOPER.md [hunk 2]
```diff
@@ -35,10 +35,60 @@ These jobs use GCP project `cloudesf-testing`.
...
+
+```
+docker run -ti --privileged -v /var/run/docker.sock:/var/run/docker.sock -v ~/esp-v2:/esp-v2 gcr.io/cloudesf-testing/gcpproxy-prow:v20230712-v2.44.0-17-g94471d2e-master  /bin/bash
```
[the remark this answers is not in the document]
Comment (siqidong) [reply DEVELOPER.md]: Added this step before running docker instance, thanks for the suggestion!

[the remark this answers is not in the document]
Comment (siqidong) [reply DEVELOPER.md]: SG! I moved the git clone to the first step to avoid confusion, both ways should work.

**Pull request merged**